ui/src/app/views/new-notebook-modal/component.ts

```diff
@@ -0,0 +1,78 @@
+import {Component, Input, OnDestroy} from '@angular/core';
+
+import {SignInService} from 'app/services/sign-in.service';
+import {Kernels} from 'app/views/notebook-redirect/component';
+
+import {environment} from 'environments/environment';
+
+import {Workspace} from 'generated';
+
+
+@Component({
+  selector: 'app-new-notebook-modal',
+  styleUrls: ['./component.css',
+    '../../styles/buttons.css',
+    '../../styles/inputs.css'],
+  templateUrl: './component.html',
+})
+export class NewNotebookModalComponent implements OnDestroy {
+  public creatingNotebook = false;
+  public newName = '';
```

Comment:
Is the new name required, or not? Note that the Jupyter UI does not require specification of a new name so we may want to match that behavior.

Reply:
It is required in this case, because we want to be able to encourage naming and renaming from our UI, to add to the cache. That is probably a Product question though. Will follow up with Karthik